Discrepancy between 5.x documentation and actual behaviour on global state restore

**Elasticsearch version** (`bin/elasticsearch --version`): 5.4.1

**Plugins installed**: []

**JVM version** (`java -version`): 1.8.0_131

**OS version** (`uname -a` if on a Unix-like system): Linux d14085ba52c1 4.4.0-66-generic #87~14.04.1-Ubuntu SMP Fri Mar 3 17:32:36 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

**Description of the problem including expected versus actual behavior**:
On version 5.4.1, when I try to restore a snapshot taken in a cluster with `minimum_master_nodes: 2` in a cluster with one instance only and `include_global_state: true`, the restore endpoint returns a 500 state and the response:
```json
{"error":{"root_cause":[{"type":"snapshot_restore_exception","reason":"[found-snapshots:scheduled-1504176332-instance-0000000002/eC4eaXFFRr2cE41gp4tQYQ] cannot restore index [.security] because it's open"}],"type":"snapshot_restore_exception","reason":"[found-snapshots:scheduled-1504176332-instance-0000000002/eC4eaXFFRr2cE41gp4tQYQ] cannot restore index [.security] because it's open"},"status":500}
```
However, according to the [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-snapshots.html), I would expect it to work and it actually does on `2.4.6`.
Should the documentation be fixed? Should the behaviour be fixed to match the docs and previous versions?

**Steps to reproduce**:

 1. create a cluster with 3 nodes and set `minimum_master_nodes` to 2
 2. create a snapshot
 3. try to restore that cluster to a 1 instance cluster with `include_global_state` true
 4. the restore endpoint should return a 500

**Provide logs (if relevant)**:
Version 5.4.1:
```logs
[2017-08-31T10:49:33,579][WARN ][org.elasticsearch.snapshots.RestoreService] [scheduled-1504176332-instance-0000000002/eC4eaXFFRr2cE41gp4tQYQ] failed to restore snapshot
java.lang.IllegalArgumentException: illegal value can't update [discovery.zen.minimum_master_nodes] from [1] to [2]
	at org.elasticsearch.common.settings.Setting$Updater.getValue(Setting.java:610) ~[elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.common.settings.AbstractScopedSettings.validateUpdate(AbstractScopedSettings.java:135) ~[elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.snapshots.RestoreService$1.restoreGlobalStateIfRequested(RestoreService.java:458) ~[elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.snapshots.RestoreService$1.execute(RestoreService.java:347) ~[elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:45) ~[elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.service.ClusterService.executeTasks(ClusterService.java:633) ~[elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.service.ClusterService.calculateTaskOutputs(ClusterService.java:611) ~[elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.service.ClusterService.runTasks(ClusterService.java:570) [elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.service.ClusterService$ClusterServiceTaskBatcher.run(ClusterService.java:262) [elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150) [elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188) [elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:569) [elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:247) [elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:210) [elasticsearch-5.4.1.jar:5.4.1]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_131]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_131]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_131]
Caused by: java.lang.IllegalArgumentException: cannot set discovery.zen.minimum_master_nodes to more than the current master nodes count [1]
	at org.elasticsearch.discovery.zen.ZenDiscovery.lambda$new$2(ZenDiscovery.java:187) ~[elasticsearch-5.4.1.jar:5.4.1]
	at org.elasticsearch.common.settings.Setting$Updater.getValue(Setting.java:608) ~[elasticsearch-5.4.1.jar:5.4.1]
	... 16 more
```

**Related PRs**
- https://github.com/elastic/elasticsearch/pull/9051
- https://github.com/elastic/elasticsearch/pull/15278

--- a/docs/reference/modules/snapshots.asciidoc
+++ b/docs/reference/modules/snapshots.asciidoc
@@ -505,15 +505,17 @@ same size or topology.  However, the version of the new cluster should be the sa
 If the new cluster has a smaller size additional considerations should be made. First of all it's necessary to make sure
 that new cluster have enough capacity to store all indices in the snapshot. It's possible to change indices settings
 during restore to reduce the number of replicas, which can help with restoring snapshots into smaller cluster. It's also
-possible to select only subset of the indices using the `indices` parameter.  Prior to version 1.5.0 elasticsearch
-didn't check restored persistent settings making it possible to accidentally restore an incompatible
-`discovery.zen.minimum_master_nodes` setting, and as a result disable a smaller cluster until the required number of
-master eligible nodes is added.  Starting with version 1.5.0 incompatible settings are ignored.
+possible to select only subset of the indices using the `indices` parameter.
 
 If indices in the original cluster were assigned to particular nodes using
 <<shard-allocation-filtering,shard allocation filtering>>, the same rules will be enforced in the new cluster. Therefore
 if the new cluster doesn't contain nodes with appropriate attributes that a restored index can be allocated on, such
 index will not be successfully restored unless these index allocation settings are changed during restore operation.
+
+The restore operation also checks that restored persistent settings are compatible with the current cluster to avoid accidentally
+restoring an incompatible settings such as `discovery.zen.minimum_master_nodes` and as a result disable a smaller cluster until the
+required number of master eligible nodes is added. If you need to restore a snapshot with incompatible persistent settings, try
+restoring it without the global cluster state.
 
 [float]
 === Snapshot status
@@ -596,7 +598,7 @@ state. Once recovery of primary shards is completed Elasticsearch is switching t
 creates the required number of replicas at this moment cluster switches to the `yellow` state. Once all required replicas
 are created, the cluster switches to the `green` states.
 
-The cluster health operation provides only a high level status of the restore process. It’s possible to get more
+The cluster health operation provides only a high level status of the restore process. It's possible to get more
 detailed insight into the current state of the recovery process by using <<indices-recovery, indices recovery>> and
 <<cat-recovery, cat recovery>> APIs.
 

